### ISSUE TITLE
Add variadic printf style for af::print

### ISSUE THREAD

**Issue by jramapuram** [arrayfire/arrayfire-rust#63]
Currently we have ugly `println!(".."); af::print(&arr);` syntax. It would be nice to have it in one func.

**Comment by jramapuram** [comment]
Not quite what I was thinking, but should be ok. Was thinking about something like: 
```rust
print("bla {:?} bla", arr); 
```

One other thing: would this macro handle the case when there is no text?
i.e:
```rust
af::print(&arr);
```

**Comment by 9prady9** [comment]
Not quite, let me see what i can do to get such behavior.

**Comment by 9prady9** [comment]
@jramapuram 

When you say, you want to do 
```rust
 print("bla {:?} bla", arr);
```
how are you expecting the format of the output would be like ?

**Comment by jramapuram** [comment]
Printf style. So the array can be inserted anywhere in the formatting string. Not only after it.

**Comment by 9prady9** [comment]
What i meant is the following.

When we usually do println!("something {:?} something", 1); what gets printed is as follows

`something 1 something". However, in case of af::Arrays, it is not a single line message, we just delegate the call to arrayfire C-API print function that prints the array in multiple lines in its own format.
So my question, we can't make it look like 

`something <Array print out> something`. 

It would probably look like the following

```
something <Array print out>
something
```
or something similar to that, but not as one would usually expect from such print statements.

**Comment by jramapuram** [comment]
Ah! The extra `\n` is not attractive. Sorry then, let's just leave this to the default behavior it is in right now without any additions!
